teachertool/src/components/makecodeFrame.tsx

```diff
@@ -0,0 +1,170 @@
+/// <reference path="../../../built/pxteditor.d.ts" />
+import * as React from "react";
+import { useState, useEffect, useRef } from "react";
+import { ProgressBar } from "react-common/components/controls/ProgressBar";
+
+// This is something I want to do eventually. The iframe should be able to work with a redux store
+// import { connect } from 'react-redux';
+
+// left  these here commented out because these states are something that might be wanted for generic cases
+// interface CloudState {
+//     [headerId: string]: pxt.cloud.CloudStatus;
+// }
+
+// interface AuthState {
+//     signedIn: boolean;
+//     profile?: pxt.auth.UserProfile;
+//     preferences?: pxt.auth.UserPreferences;
+// }
+// export const cloudLocalStoreKey = "-SHOWN-LOGIN-PROMPT";
+
+
+interface MakeCodeFrameProps {
+    pageSourceUrl: string;
+    highContrast?: boolean;
+    onFrameOpen: () => void;
+    onFrameClose: () => void;
+    tutorialEventHandler?: (event: pxt.editor.EditorMessageTutorialEventRequest) => void;
+    // TODO: make it so the iframe can work with a redux store
+}
+
+type FrameState = "loading" | "no-project" | "opening-project" | "project-open" | "closing-project";
+
+interface PendingMessage {
+    original: pxt.editor.EditorMessageRequest;
+    handler: (response: any) => void;
+}
+
+export const MakeCodeFrame: React.FC<MakeCodeFrameProps> =
+    ( { pageSourceUrl,
+        highContrast,
+        onFrameClose,
+        onFrameOpen
+    } ) => {
+    let ref: HTMLIFrameElement | undefined;
+    const messageQueue: pxt.editor.EditorMessageRequest[] = [];
+    let nextId: number = 0;
+    let pendingMessages: {[index: string]: PendingMessage} = {};
+    const [frameState, setFrameState] = useState<FrameState>("loading");
+    const [loadPercent, setLoadPercent] = useState(0);
+    const [workspaceReady, setWorkspaceReady] = useState(false);
+
+    let frameRef = useRef<HTMLIFrameElement>(null);
+
+    useEffect(() => {
+        // logic we want to do when the iframe is loaded
+        const root = document.getElementById("root");
+        if (ref && ref.contentWindow) {
+            window.addEventListener("message", onMessageReceived);
+            ref.addEventListener("load", handleFrameReload)
+
+            if (root) pxt.BrowserUtils.addClass(root, "editor");
```

Comment:
What's the motivation for adding the editor class to our root element?

Reply:
Ah, good question. I thought this was important, but investigated and it looks like this was added for skillmap to work around some weirdness with the header, but I don't think that will apply in this situation. If there are special cases like this, I think it's something we should address in the future, so I'll take it out for now. Thanks, Thomas!